pkg/object/ingresscontroller/translator.go

```diff
@@ -379,6 +397,25 @@ func (st *specTranslator) translate() error {
 		})
 	}
 
+	// sort path:
+	// * precise path first
+	// * longer prefix first
+	for _, r := range b.Rules {
+		sort.Slice(r.Paths, func(i, j int) bool {
+			p1, p2 := r.Paths[i], r.Paths[j]
+			switch {
+			case p1.Path == "" && p2.Path == "":
+				return p1.Path < p2.Path
```

Comment:
Isn't this always false, as `p1.Path` and `p2.Path` are both empty strings `""`, so p2 cannot be larger?

Reply:
I would consider moving this sorting function to `util` package and adding testsuite for it.

Reply:
my mistake, it should be `!=`.